### PR TITLE
Fix get_training_or_validation_split returning empty training set when validation rounds to 0

### PR DESCRIPTION
**Pull request by Osamaali313**
### Summary

`get_training_or_validation_split` (in `keras/src/utils/dataset_utils.py`) splits a list of samples/labels into a training or validation subset. It computes `num_val_samples = int(validation_split * len(samples))` and then slices with **negative** indices:

```python
samples = samples[:-num_val_samples]   # training
samples = samples[-num_val_samples:]   # validation
```

When `int(validation_split * len(samples))` rounds down to **0** — a small dataset and/or a small `validation_split` — these become `samples[:-0]` (== `samples[:0]`, **empty**) and `samples[-0:]` (== `samples[0:]`, **everything**). So:

- the **training** subset comes back **empty** (while `print_msg` claims *"Using N files for training."*), and
- the **validation** subset wrongly contains **all** the samples.

This is reachable through the public `image_dataset_from_directory` / `text_dataset_from_directory` / `audio_dataset_from_directory` APIs (`check_validation_split_arg` only enforces `0 < validation_split < 1`). The empty training split then trips a misleading `ValueError("No images found in directory ...")` even though files exist.

**Before** (10 samples, `validation_split=0.05` → `num_val_samples = int(0.5) = 0`):

```
subset="training"   -> []          (logged "Using 10 files for training.")
subset="validation" -> [0..9]      (all 10)
```

### Fix

Compute `num_train_samples = len(samples) - num_val_samples` and slice with positive indices, which avoids the negative-zero slice in both branches:

```python
num_train_samples = len(samples) - num_val_samples
# training:   samples[:num_train_samples]
# validation: samples[num_train_samples:]
```

**After:**

```
subset="training"   -> [0..9]      (all 10, matches the log)
subset="validation" -> []          (correctly empty)
# validation_split=0.2 unchanged: training [0..7], validation [8, 9]
```

### Tests

Added `test_get_training_or_validation_split_small_validation_split` (the rounds-to-zero case) and `test_get_training_or_validation_split_regular` to `keras/src/utils/dataset_utils_test.py`. The first fails on the current code and passes with the fix; `ruff check`/`ruff format` are clean.

### AI-Assisted Contribution Policy disclosure

Per the policy: AI assistance was used to help **discover** this edge case and to **verify** it (reproducing the empty-split behavior and cross-checking the slice arithmetic). I reviewed and fully understand the bug and the fix, wrote/tested the change myself, and take full responsibility for it.

### CODE REVIEW

**Review by hertschuh — APPROVED** [review]
Thanks for the fix!
